aws/resource_aws_iam_service_specific_credential.go

```diff
@@ -0,0 +1,184 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/iam"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func resourceAwsIamServiceSpecificCredential() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsIamServiceSpecificCredentialCreate,
+		Read:   resourceAwsIamServiceSpecificCredentialRead,
+		Update: resourceAwsIamServiceSpecificCredentialUpdate,
+		Delete: resourceAwsIamServiceSpecificCredentialDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"service_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"user_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringLenBetween(1, 64),
+			},
+			"status": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Default:      iam.StatusTypeActive,
+				ValidateFunc: validation.StringInSlice(iam.StatusType_Values(), false),
+			},
+			"service_password": {
+				Type:      schema.TypeString,
+				Sensitive: true,
+				Computed:  true,
+			},
+			"service_user_name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"service_specific_credential_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceAwsIamServiceSpecificCredentialCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).iamconn
+
+	input := &iam.CreateServiceSpecificCredentialInput{
+		ServiceName: aws.String(d.Get("service_name").(string)),
+		UserName:    aws.String(d.Get("user_name").(string)),
+	}
+
+	out, err := conn.CreateServiceSpecificCredential(input)
+	if err != nil {
+		return fmt.Errorf("error creating IAM Service Specific Credential: %w", err)
+	}
+
+	cred := out.ServiceSpecificCredential
+
+	d.SetId(fmt.Sprintf("%s:%s", aws.StringValue(cred.ServiceName), aws.StringValue(cred.UserName)))
+	d.Set("service_password", cred.ServicePassword)
+
+	if v, ok := d.GetOk("status"); ok && v.(string) == iam.StatusTypeInactive {
```

Comment:
I would switch from `v.(string) == iam.StatusTypeInactive` to `v.(string) != iam.StatusTypeActive`, so that if AWS adds more statuses we'd be covered (that seems more likely than them changing the default status to `Active`).

Reply:
agree, added.